gateway/src/apicast/executor.lua

```diff
@@ -59,6 +61,45 @@ function _M:context(phase)
     return shared_build_context(self)
 end
 
+do
+    local policy_loader = require('apicast.policy_loader')
+    local policies
+
+    local init = _M.init
+    function _M:init()
+        local executed = {}
+
+        for _,policy in init(self) do
+            executed[policy.init] = true
+        end
+
+        policies = policy_loader:get_all()
+
+        for _, policy in ipairs(policies) do
+            if not executed[policy.init] then
+                policy.init()
+                executed[policy.init] = true
+            end
+        end
+    end
+
+    local init_worker = _M.init_worker
+    function _M:init_worker()
+        local executed = {}
+
+        for _,policy in init_worker(self) do
+            executed[policy.init_worker] = true
+        end
+
+        for _, policy in ipairs(policies or policy_loader:get_all()) do
```

Comment:
This caching complicates a lot testing this method. Depending on the execution the value will be cached or not and when it is, mocking `policy_loader:get_all()` will have no effect.
The tests I added in my latest commit are affected by this.

Reply:
Are you ok with not caching this @mikz ?

Reply:
I see. It would make more sense to me to provide reset for the cache than not having it at all. 

I think it is important to be correct and that init and init_worker should be executed on exactly the same lists. Loading the list for the second time could not ensure that.

Reply:
:+1: